server/liveblog/themes/themes.py

```diff
@@ -81,6 +83,23 @@ def update_registered_theme_with_local_files(self):
                 created.append(theme)
         return (created, updated)
 
+    def on_delete(self, doc):
+        blogs_service = get_resource_service('blogs')
+        blogs = blogs_service.get(req=None, lookup=dict(blog_status='open'))
```

Comment:
You should add in the lookup the removed theme to filter blogs, then you don't have to loop all over the blogs after to look after the ones that use the theme.